### PR TITLE
feat: run a check suite on a PR

### PR DESCRIPTION
**Pull request by technosophos**
This converts a PR request to a check suite request by notifying the GitHub gateway to create a check suite.

### CODE REVIEW

File: charts/brigade-github-app/values.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ rbac:
   enabled: false
 
 # Image configuration
-registry: technosophos
+registry: deis
```
Comment (vdice) [inline charts/brigade-github-app/values.yaml]: 👍 -- there may be another instance, I think in docs/README.md still referencing `technosophos`

Comment (technosophos) [reply charts/brigade-github-app/values.yaml]: Fixed

File: README.md [hunk 2]
```diff
@@ -141,9 +141,29 @@ You will want to make sure to set:
 - `sharedSecret` to use the shared secret you created when creating the app
 - `github.token` (aka `github: {token: }`) to the OAuth token GitHub Apps gave you
 
+## 7. (OPTIONAL): Forwarding `pull_request` to `check_suite`
+
+This gateway can enable a feature that converts PRs to Check Suite requests. Currently, this is enabled bu default. To disable
```
[the remark this answers is not in the document]
Comment (technosophos) [reply README.md]: fixed

File: cmd/github-gateway/server.go [hunk 3]
```diff
@@ -65,6 +66,26 @@ func main() {
...
+		if err != nil {
+			return defaultVal
+		}
+		return realVal
+	}
+
+	csopr := envOrInt("CHECK_SUITE_ON_PR", 1) == 1
+	ghOpts := webhook.GithubOpts{
+		CheckSuiteOnPR: csopr,
+		AppID:          envOrInt("APP_ID", 0),          //20265,
```
[the remark this answers is not in the document]
Comment (technosophos) [reply cmd/github-gateway/server.go]: Ugh... glad you caught that. Thanks

File: pkg/webhook/github.go [hunk 4]
```diff
@@ -275,10 +315,103 @@ func (s *githubHook) handleEvent(c *gin.Context, eventType string) {
...
+//	- try to create a check_suite
+//		- if success, run a `rerequest` on this check suite because merely creating a check suite does
+// 		  not actually trigger a check_suite:requested webhook event
+//		- if failure, check to see if we already have a check suite object, and merely run the rerequest
+//		  on that check suite.
+func (s *githubHook) prToCheckSuite(c *gin.Context, pre *github.PullRequestEvent, proj *brigade.Project) error {
+	repo := pre.Repo.GetFullName()
+	ref := pre.PullRequest.Head.GetRef()
+	sha := pre.PullRequest.Head.GetSHA()
+	appID := s.opts.AppID           //pre.Installation.GetAppID()
```
[the remark this answers is not in the document]
Comment (technosophos) [reply pkg/webhook/github.go]: So the documentation says these values can be there. But testing seems to indicate that they are not. So I will just remove the commented lines.

File: README.md [hunk 5]
```diff
@@ -141,9 +141,29 @@ You will want to make sure to set:
...
+this feature, set the environment variable `CHECK_SUITE_ON_PR=0` on the deployment for the server. TODO: add this to chart.
```
[the remark this answers is not in the document]
Comment (technosophos) [reply README.md]: Thank you!

**Pull request merged**